apps/speed.c

```diff
@@ -1916,6 +1929,32 @@ int speed_main(int argc, char **argv)
             }
         }
     }
+    /* repeated code good to factorize */
+    ecdh_c[R_EC_BRP256R1][0] = count / 1000;
+    for (i = R_EC_BRP384R1; i < R_EC_BRP512R1; i += 2) {
+        ecdh_c[i][0] = ecdh_c[i - 2][0] / 2;
+        if (ecdh_doit[i] <= 1 && ecdh_c[i][0] == 0)
+            ecdh_doit[i] = 0;
+        else {
+            if (ecdh_c[i][0] == 0) {
+                ecdh_c[i][0] = 1;
+            }
+        }
+    }
+    ecdh_c[R_EC_BRP256T1][0] = count / 1000;
+    for (i = R_EC_BRP384T1; i < R_EC_BRP512T1; i += 2) {
+        ecdh_c[i][0] = ecdh_c[i - 2][0] / 2;
+        if (ecdh_doit[i] <= 1 && ecdh_c[i][0] == 0)
+            ecdh_doit[i] = 0;
+        else {
+            if (ecdh_c[i][0] == 0) {
+                ecdh_c[i][0] = 1;
+            }
+        }
+    }
+    /* default iteration count for the last two EC Curves */
+    ecdh_c[R_EC_X25519][0] = count / 2200;
+    ecdh_c[R_EC_X448][0] = count / 800;
```

Comment:
Suggestion was that x448 coefficient would be 8 times larger than x25519. I.e. if X25519 was 1000, then X448 would be 8000. Slower algorithm -> higher coefficient. X448 is slower than X25519, 8 times (on x86_64). Well, one can argue that it's not really fair comparison, and it should be approximately square of key lengths' ratio, all right, let's say 4. So if you choose 2200 for X25519, then it should be 8800 for X448.

Reply:
I made some measurements of X25519/X499 against ecdhb163 to deduce these ratios, but I inverse my calculation Oups ;(
I get a x3 ratio on my computer between  X25519 and X499, but ok for a x4 ratio.